staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -229,6 +231,11 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	crdName := requestInfo.Resource + "." + requestInfo.APIGroup
 	crd, err := r.crdLister.Get(crdName)
 	if apierrors.IsNotFound(err) {
+		if !r.hasSynced() {
+			http.Error(w, "server is starting", http.StatusServiceUnavailable)
```

Comment:
Shouldn’t we return a structured error with a retry after 10s?

Reply:
switched to structured error with timeout